src/version/set.rs

```diff
@@ -81,16 +81,39 @@ where
         clean_sender: Sender<CleanTag>,
         option: Arc<DbOption<R>>,
     ) -> Result<Self, VersionError<R>> {
-        let (mut log, log_id) = if let Some(file) = pin!(FP::list(
+        let mut log_stream = pin!(FP::list(
             option.version_log_dir_path(),
             FileType::Log,
             true
-        )?)
-        .next()
-        .await
-        .transpose()?
-        {
-            file
+        )?);
+        let mut first_log_id = None;
+        let mut version_log_id = None;
+        let mut version_log = None;
+
+        // when there are multiple logs, this means that a downtime occurred during the
+        // `version_log_snap_shot` process, the second newest file has the highest data
+        // integrity, so it is used as the version log, and the older log is deleted first
+        // to avoid midway downtime, which will cause the second newest file to become the
+        // first newest after restart.
+        let mut i = 0;
+        while let Some(result) = log_stream.next().await {
+            let (log, log_id) = result?;
+
+            if i <= 1 {
```

Comment:
`list` sort files in increasing order, so the last one is the newest file?
https://github.com/tonbo-io/tonbo/blob/f4ab2e54af749181a2dc2f104189dd56854b9c5a/src/fs/tokio_fs.rs#L40

Reply:
yep, so I reversed it
https://github.com/tonbo-io/tonbo/blob/f4ab2e54af749181a2dc2f104189dd56854b9c5a/src/version/set.rs#L87

Reply:
Oops, didn't notice that. But it seems that `Ulid::new` will not guarantee monotonic sort order. maybe use `ulid::Generator` instead?
https://docs.rs/ulid/latest/ulid/struct.Ulid.html#method.new

Reply:
You are right, we should be able to open this as a new issue